pkg/epp/util/request/body_test.go

```diff
@@ -39,169 +52,369 @@ func TestExtractPromptFromRequestBody(t *testing.T) {
+	switch {
+	case got.Completions != nil && want.Completions != nil:
+		return compareCompletionsRequest(got.Completions, want.Completions, t)
+	case got.ChatCompletions != nil && want.ChatCompletions != nil:
+		return compareChatCompletionsRequest(got.ChatCompletions, want.ChatCompletions, t)
+	case got.Completions == nil && want.Completions == nil && got.ChatCompletions == nil && want.ChatCompletions == nil:
+		return true
+	default:
+		t.Errorf("Result type mismatch: got completions=%v, chatCompletions=%v; want completions=%v, chatCompletions=%v",
+			got.Completions != nil, got.ChatCompletions != nil, want.Completions != nil, want.ChatCompletions != nil)
+		return false
 	}
+}
 
-	for _, tt := range tests {
-		if got := constructChatMessage(tt.role, tt.content); got != tt.want {
-			t.Errorf("constructChatMessage() = %v, want %v", got, tt.want)
+func compareCompletionsRequest(got, want *types.CompletionsRequest, t *testing.T) bool {
+	if got.Prompt != want.Prompt {
+		t.Errorf("CompletionsRequest.Prompt = %v, want %v", got.Prompt, want.Prompt)
+		return false
+	}
+	return true
+}
+
+func compareChatCompletionsRequest(got, want *types.ChatCompletionsRequest, t *testing.T) bool {
+	// Compare messages
+	if len(got.Messages) != len(want.Messages) {
+		t.Errorf("Messages length = %v, want %v", len(got.Messages), len(want.Messages))
+		return false
+	}
+	for i, msg := range got.Messages {
+		wantMsg := want.Messages[i]
+		if msg.Role != wantMsg.Role || msg.Content != wantMsg.Content {
+			t.Errorf("Message[%d] = %v, want %v", i, msg, wantMsg)
+			return false
+		}
+	}
+
+	// Compare optional fields
+	if got.ChatTemplate != want.ChatTemplate {
+		t.Errorf("ChatTemplate = %v, want %v", got.ChatTemplate, want.ChatTemplate)
+		return false
+	}
+	if got.ReturnAssistantTokensMask != want.ReturnAssistantTokensMask {
+		t.Errorf("ReturnAssistantTokensMask = %v, want %v", got.ReturnAssistantTokensMask, want.ReturnAssistantTokensMask)
+		return false
+	}
+	if got.ContinueFinalMessage != want.ContinueFinalMessage {
+		t.Errorf("ContinueFinalMessage = %v, want %v", got.ContinueFinalMessage, want.ContinueFinalMessage)
+		return false
+	}
+	if got.AddGenerationPrompt != want.AddGenerationPrompt {
+		t.Errorf("AddGenerationPrompt = %v, want %v", got.AddGenerationPrompt, want.AddGenerationPrompt)
+		return false
+	}
+
+	// Compare tools (shallow comparison for test purposes)
+	if !compareSliceAny(got.Tools, want.Tools) {
+		t.Errorf("Tools mismatch")
+		return false
+	}
+
+	// Compare documents (shallow comparison for test purposes)
+	if !compareSliceAny(got.Documents, want.Documents) {
+		t.Errorf("Documents mismatch")
+		return false
+	}
+
+	// Compare chat template kwargs (shallow comparison for test purposes)
+	if !compareMapAny(got.ChatTemplateKWArgs, want.ChatTemplateKWArgs) {
+		t.Errorf("ChatTemplateKWArgs mismatch")
+		return false
+	}
+
+	return true
+}
+
+func compareSliceAny(got, want []any) bool {
```

Comment:
any reason we need a custom compare instead of `cmp.Diff`? Same comment for the map comparison